### ISSUE TITLE
The fallback function declared after the receive function shouldn't trigger the ordering rule

### ISSUE THREAD

**Issue by ilkecan** [protofire/solhint#248]
[ordering](https://github.com/protofire/solhint/blob/master/docs/rules/order/ordering.md) rule shows false positive when there is an `external receive` function before the `fallback` functions. For example, for this contract:
```
contract Test {
    receive() external payable {
    }

    fallback() external payable {
    }
}
```
Solhint generates this warning:
`Function order is incorrect, fallback function can not go after external function`
which is incorrect, according to the [Style Guide](https://solidity.readthedocs.io/en/v0.7.1/style-guide.html#order-of-functions).
